### PR TITLE
Cartify implementation  

### PR DESCRIPTION
**Pull request by shahin1290**
## Pivotal Tracker link
[Link to PT](https://www.pivotaltracker.com/n/projects/2227327)

## Actions taken
Created A feature file -> cartify_implemented.feature
Added feature, user story and scenario
Added cartify gem
Added 'add to cart' button to each item, cartify helper methods to that button
screenshot from 2018-11-30 16-19-46
screenshot from 2018-11-30 16-20-00
## What we have learned
How to implement cartify
How to use dom_id helper and find content within content

### CODE REVIEW

File: features/cartify_implemented.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Feature: Visitor can add dishes to an order
+    As a visitor,
+    In order to select dishes I want to buy
+    I would like to be able to add dishes to an order
+
+    @javascript
+    Scenario: Visitor can add dishes to an order
+        Given the following dishes exist
```
Comment (oliverochman) [inline features/cartify_implemented.feature:7]: Put this table in a background

File: db/schema.rb [hunk 2]
```diff
@@ -10,11 +10,76 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2018_11_28_124202) do
+ActiveRecord::Schema.define(version: 2018_11_28_192230) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "cartify_addresses", force: :cascade do |t|
```
Comment (oliverochman) [inline db/schema.rb:18]: I want to see unit tests for these tables

File: app/views/dishes/index.html.haml [hunk 3]
```diff
@@ -5,19 +5,23 @@
     - else
```
Comment (oliverochman) [inline app/views/dishes/index.html.haml]: extract the login and sign up to a partial

File: app/models/user.rb [hunk 4]
```diff
@@ -6,4 +6,9 @@ class User < ApplicationRecord
 
   validates_presence_of :name, :phone, :address
 
+  has_many :orders, class_name: 'Cartify::Order', foreign_key: :user_id
```
Comment (oliverochman) [inline app/models/user.rb:9]: update the user model specs to test these associations

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/cartify_implemented.feature (hunk 1 above)
Comment (shahin1290) [reply features/cartify_implemented.feature:7]: done

File: db/schema.rb (hunk 2 above)
Comment (shahin1290) [reply db/schema.rb:18]: done

File: app/views/dishes/index.html.haml (hunk 3 above)
Comment (shahin1290) [reply app/views/dishes/index.html.haml]: done

File: app/models/user.rb (hunk 4 above)
Comment (shahin1290) [reply app/models/user.rb:9]: done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**